variables.tf

```diff
@@ -181,9 +181,32 @@ variable "auto_scaling" {
   })
   description = "(Optional) Configure rules to allow your database to automatically increase its resources. Single block of autoscaling is allowed at once."
   default = {
-    cpu    = {}
-    disk   = {}
-    memory = {}
+    cpu = {
+      rate_increase_percent       = 20
+      rate_limit_count_per_member = 20
+      rate_period_seconds         = 900
+      rate_units                  = "count"
+    }
+    disk = {
+      capacity_enabled             = true
+      free_space_less_than_percent = 10
+      io_above_percent             = 90
+      io_enabled                   = true
+      io_over_period               = "15m"
+      rate_increase_percent        = 10
+      rate_limit_mb_per_member     = 3670016
+      rate_period_seconds          = 900
+      rate_units                   = "mb"
+    }
+    memory = {
+      io_above_percent         = 90
+      io_enabled               = true
+      io_over_period           = "15m"
+      rate_increase_percent    = 10
+      rate_limit_mb_per_member = 114688
+      rate_period_seconds      = 900
+      rate_units               = "mb"
+    }
```

Comment:
This is insufficient to prevent consumers encountering issues at apply time. Possibly issue with 1.50.0 provider, see https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4339

One good example of this is that most of the examples fail with a v1.51 provider, because they over ride this new default behaviour.

There is no technical reason to upgrade the provider to v1.50+ for this component, so it should not be upgraded. However, consumers may use higher versions and may encounter the regression (which is the issue this PR is addressing) so it should be validated manually.

Some decisions are required to determine the best solution.
Does autoscaling get enabled even if the block is not specified? The answer to this question might help shape the best solution.
- Should the optional properties be mandatory? At least some of them are required with the latest providers, maybe all of them.
- Should the consumer be allowed take the provider default values? This could be implemented by setting all the default values, but the provider could change these at any time. Adding a dynamic construct around the auto_scaling block in the main module enables it to be excluded, taking the provider generated default values.

### Output from manual testing with v1.51 provider

Note: The provision took significantly longer than expected before it failed with the error message.

    module.postgresql_db.ibm_database.postgresql_db: Still creating... [47m10s elapsed]
    ╷
    │ Error: [ERROR] Error updating database auto_scaling: Unprocessable Entity
    │ 
    │   with module.postgresql_db.ibm_database.postgresql_db,
    │   on ../../main.tf line 12, in resource "ibm_database" "postgresql_db":
    │   12: resource "ibm_database" "postgresql_db" {
    │ 
    ╵

The only change made was


    git diff
    diff --git a/examples/complete/version.tf b/examples/complete/version.tf
    index b672faa..38d0fca 100644
    --- a/examples/complete/version.tf
    +++ b/examples/complete/version.tf
    @@ -4,7 +4,7 @@ terraform {
         # Pin to the lowest provider version of the range defined in the main module's version.tf to ensure lowest version still works
         ibm = {
           source  = "IBM-Cloud/ibm"
    -      version = "1.49.0"
    +      version = "1.51.0"
         }
       }
     }